### DISCUSSION TITLE
TRSP from / to a two way road

### DISCUSSION THREAD

**Discussion by salvatierra** [pgRouting/pgrouting#2409] [Q&A]
In pgRouting 3.3 function pgr_trsp it was possible to specify the position on the start / end edge instead of just the start / end vertex. This seemed very useful since if the start / end edge is a two-way edge and the source / target was a point in the middle of the edge (as in most real life routing problems) there is no way of knowing from which vertex it's better to "enter" / "exit" the edge but comparing the cost of the possible routes. Why was this deprecated? Is there another alternative than to produce various routes?

**Comment by cvvergara** [comment]
Have you tried [pgr_trsp_withPoints](https://docs.pgrouting.org/latest/en/pgr_trsp_withPoints.html)
We have a [migration guide](https://docs.pgrouting.org/latest/en/migration.html)
